hugging_face_classifier_trainer/hugging_face_classifier_trainer.py

```diff
@@ -0,0 +1,794 @@
+        train_dataset = train_dataset.shuffle(seed=random_state).select(
+            list(range(num_of_train_samples))
+        )
+    train_dataset = _edit_columns(train_dataset, drop_columns, rename_cols)
+
+    # test set
+    test_dataset = dataset["test"]
+    if train_test_split_size or num_of_train_samples:
+        train_test_split_size = train_test_split_size or 0.2
+        num_of_test_samples = int(
+            (train_dataset.num_rows * train_test_split_size)
+            // (1 - train_test_split_size)
+        )
+        test_dataset = test_dataset.shuffle(seed=random_state).select(
+            list(range(num_of_test_samples))
+        )
+    test_dataset = _edit_columns(test_dataset, drop_columns, rename_cols)
+
+    return train_dataset, test_dataset
+
+
+def train(
+    context: MLClientCtx,
+    hf_dataset: str = None,
+    dataset: DataItem = None,
+    drop_columns: Optional[List[str]] = None,
+    pretrained_tokenizer: str = None,
+    pretrained_model: str = None,
+    model_class: str = None,
+    model_name: str = "huggingface-model",
+    label_name: str = "labels",
+    text_col: str = "text",
+    num_of_train_samples: int = None,
+    train_test_split_size: float = None,
+    metrics: List[str] = None,
+    random_state: int = None,
+):
+    """
+    Training and evaluating a pretrained model with a pretrained tokenizer over a dataset.
+    The dataset can be either be the name of the dataset that contains in the HuggingFace hub,
+    or a URI or a FeatureVector
+
+    :param context:                 MLRun context
+    :param hf_dataset:              The name of the dataset to get from the HuggingFace hub
+    :param dataset:                 The dataset to train the model on. Can be either a URI or a FeatureVector
+    :param drop_columns:            The columns to drop from the dataset.
+    :param pretrained_tokenizer:    The name of the pretrained tokenizer from the HuggingFace hub.
+    :param pretrained_model:        The name of the pretrained model from the HuggingFace hub.
+    :param model_name:              The model's name to use for storing the model artifact, default to 'model'
+    :param model_class:             The class of the model, e.g. `transformers.AutoModelForSequenceClassification`
+    :param label_name:              The target label of the column in the dataset.
+    :param text_col:                The input text column un the dataset.
+    :param num_of_train_samples:    Max number of training samples, for debugging.
+    :param train_test_split_size:   Should be between 0.0 and 1.0 and represent the proportion of the dataset to include
+                                    in the test split.
+    :param metrics:                 List of different metrics for evaluate the model such as f1, accuracy etc.
+    :param random_state:            Random state for train_test_split
+    """
+
+    if not label_name or not pretrained_tokenizer:
+        raise mlrun.errors.MLRunRuntimeError(
```

Comment:
I think `MLRunInvalidArgumentError` is more fitting.

Also, if they are mandatory, why they are optional in the func header? (`= None`) Simply make them mandatory and you can remove this validation.